### PR TITLE
[shopsys] encapsulation of collections inside of entities

### PR DESCRIPTION
**Pull request by pk16011990**
| Q             | A
| ------------- | ---
|Description, reason for the PR| https://ocramius.github.io/doctrine-best-practices/#/43
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| Yes 
|Fixes issues| ... 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by vitek-rostislav** [comment]
Hi @pk16011990, thanks for the PR. My apologies for the very late response. Processing of your PR is in our buffer, however, there are some other priorities in our sprint so it might take a while.

**Comment by boris-brtan** [comment]
what do you think of this PR @vitek-rostislav ???

**Comment by boris-brtan** [comment]
@pk16011990 if there is some change in frameworkBundle that needs changes in shopBundle tests it is considered as BC-break, so this should be pointed into 8.0

**Comment by boris-brtan** [comment]
all bc-breaks should be based on new MAJOR version branch. In this case `8.0`
rebased on 8.0 resolved conflicts, added more annotation fixes

**Comment by pk16011990** [comment]
I would like to ask you for a favor. Can you please finish this PR without me? After a long pause there were conflicts and I have no guarantee that I will solve it and there will be new conflicts before you look again after XY days.

I understand that it can delay the merge, but I'm OK with it.

File: packages/framework/src/Model/Administrator/Administrator.php [hunk 1]
```diff
@@ -61,7 +61,8 @@ class Administrator implements UserInterface, Serializable, UniqueLoginInterface
     protected $email;
 
     /**
-     * @var \Doctrine\Common\Collections\ArrayCollection|\Shopsys\FrameworkBundle\Model\Administrator\AdministratorGridLimit[]
+     * @var \Shopsys\FrameworkBundle\Model\Administrator\AdministratorGridLimit[]|\Doctrine\Common\Collections\ArrayCollection
```
Comment (simara-svatopluk) [inline packages/framework/src/Model/Administrator/Administrator.php]:
```suggestion
     * @var \Shopsys\FrameworkBundle\Model\Administrator\AdministratorGridLimit[]|\Doctrine\Common\Collections\Collection
```

Initially the `ArrayCollection` is true, but once the Doctrine hydrates the entity, it hydrates the `$gridLimit` as a `PersistentCollection`, and these two classes are different. The only think that they have in common is the interface `Collection`

Comment (simara-svatopluk) [reply packages/framework/src/Model/Administrator/Administrator.php]: Please change it in all occurences in this PR

File: project-base/src/Shopsys/ShopBundle/Form/Front/Order/TransportAndPaymentFormType.php [hunk 2]
```diff
@@ -95,7 +95,7 @@ public function validateTransportPaymentRelation(OrderData $orderData, Execution
 
         $relationExists = false;
         if ($payment instanceof Payment && $transport instanceof Transport) {
-            if ($payment->getTransports()->contains($transport)) {
+            if (in_array($transport, $payment->getTransports(), true) === true) {
```
Comment (simara-svatopluk) [inline project-base/src/Shopsys/ShopBundle/Form/Front/Order/TransportAndPaymentFormType.php]:
```suggestion
            if (in_array($transport, $payment->getTransports(), true)) {
```

**Comment by boris-brtan** [comment]
the whole day is github freezed and buggy https://travis-ci.org/shopsys/shopsys/builds/551349551

**Comment by TomasLudvik** [comment]
I think that we should mention, that return values of some framework methods has changed - what do you think guys? @simara-svatopluk @boris-brtan @pk16011990

**Comment by simara-svatopluk** [comment]
Please change the commit message (and then I approve the PR). Tomas is right - there are changes in return values and upgrade notes should reflect it

**Comment by boris-brtan** [comment]
commimt message changed and upgrade notes updated

**Pull request merged**